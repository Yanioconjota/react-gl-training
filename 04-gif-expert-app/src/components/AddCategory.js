import React, { useState } from "react";
import Button from 'react-bootstrap/Button';
import PropTypes from "prop-types";

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('');
  
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const eventValue = inputValue;
    
    if (eventValue.trim() !== "") {
      setCategories(() => [
       inputValue,
      ]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <div className="input-group mb-3">
        <input
          type="text"
          placeholder="Type something"
          onChange={handleInputChange}
          className="form-control"
          value={inputValue}
        />
        <Button type="submit" variant="secondary">
          Agregar
        </Button>
      </div>
    </form>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};

export default AddCategory;