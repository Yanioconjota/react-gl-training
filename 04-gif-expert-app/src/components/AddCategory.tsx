import React, { useState } from "react";
import Button from 'react-bootstrap/Button';
import PropTypes from "prop-types";

export const AddCategory = ({setCategories, categories}) => {

  const [inputValue, setInputValue] = useState('');
  
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
    console.log('handleInputChange triggered')
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('handleSubmit');
    const eventValue = inputValue;
    
    if (eventValue.trim() !== "") {
      //setCategories(898); para probar que se haya llamado con una funciÃ³n
      setCategories(() => [
       inputValue,
      ]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <div className="input-group mb-3">
        <input
          type="text"
          placeholder="Type something"
          onChange={handleInputChange}
          className="form-control"
          value={inputValue}
        />
        <Button type="submit" variant="secondary">
          Agregar
        </Button>
      </div>
    </form>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};

export default AddCategory;