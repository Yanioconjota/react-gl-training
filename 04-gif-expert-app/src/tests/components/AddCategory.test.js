import "@testing-library/jest-dom";
import { mount, shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Pruebas en AddCategory", () => {
  const setCategories = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategories}/>);

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
