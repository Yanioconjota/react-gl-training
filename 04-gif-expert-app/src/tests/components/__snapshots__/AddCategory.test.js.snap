// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-group mb-3"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Type something"
      type="text"
      value=""
    />
    <Button
      active={false}
      disabled={false}
      type="submit"
      variant="secondary"
    >
      Agregar
    </Button>
  </div>
</form>
`;
